src/gui/folder.cpp

```diff
@@ -314,6 +314,62 @@ void Folder::etagRetreivedFromSyncEngine(const QString &etag)
     _lastEtag = etag;
 }
 
+void Folder::slotAccountCapabilitiesChanged()
+{
+    const Capabilities& capabilities = _accountState->account()->capabilities();
+    QString wsUrl = capabilities.getWebSocketUrl();
+    if (wsUrl.isEmpty()) {
+        return;
+    }
+
+    if (_webSocket) {
+        return;
+    }
+    //initialize wamp connection
+    _webSocket.reset(new QTcpSocket());
+    QUrl webSocketUrl(wsUrl);
+    QObject::connect(_webSocket.data(), &QTcpSocket::connected, [&]() {
+
+        QString sessionName("sessionName");
+        _session.reset(new QWamp::Session(sessionName, *_webSocket.data(), QWamp::Session::MessageFormat::Msgpack, true));
+
+        QObject::connect(_session.data(), &QWamp::Session::joined, [&](qint64 s) {
+            qDebug() << "Session joined to realm1 with session ID " << s;
+
+            // TODO: add current userid
+            _session->subscribe("files.root-etag-changed.admin", [&](const QVariantList& args, const QVariantMap& options) {
+                qDebug() << "Event received";
+                QString notificationEtag = args.at(0).toString();
+                QString notificationUser = args.at(1).toString();
+                QString user = _accountState->account()->credentials()->user();
+                if (user == notificationUser && _lastEtag != notificationEtag) {
+                    qDebug() << "* [WebSocket] Compare etag with previous etag: last:" << _lastEtag << ", received:" << notificationEtag << "-> CHANGED";
+                    _lastEtag = notificationEtag;
+                    slotScheduleThisFolder();
+                    _accountState->tagLastSuccessfullETagRequest();
+                }
+            });
+        });
+
+        QObject::connect(_session.data(), &QWamp::Session::started, [&]() {
+            _session->join("realm1");
+        });
+        _session->start();
+    });
+
+    QObject::connect(_webSocket.data(), static_cast<void (QAbstractSocket::*)(QAbstractSocket::SocketError err)>(&QAbstractSocket::error), [&](QAbstractSocket::SocketError err) {
+        qInfo() << "tcp error: " << err;
```

Comment:
This probably deserve to be a qCWarning. Don't forget to add a category.  Also keep it in one line.